algorithms/src/snark/marlin/ahp/prover/state.rs

```diff
@@ -114,13 +114,13 @@ impl<'a, F: PrimeField, MM: MarlinMode> State<'a, F, MM> {
 
                 let first_padded_public_inputs = &variable_assignments[0].0;
                 let input_domain = EvaluationDomain::new(first_padded_public_inputs.len()).unwrap();
-                let batch_size = variable_assignments.len();
+                let batch_size = u32::try_from(variable_assignments.len())?;
                 total_instances += batch_size;
-                let mut z_as = Vec::with_capacity(batch_size);
-                let mut z_bs = Vec::with_capacity(batch_size);
-                let mut x_polys = Vec::with_capacity(batch_size);
-                let mut padded_public_variables = Vec::with_capacity(batch_size);
-                let mut private_variables = Vec::with_capacity(batch_size);
+                let mut z_as = Vec::with_capacity(variable_assignments.len());
```

Comment:
Maybe we should make a `batch_size_usize` local variable and use that here?

Reply:
Indeed would be easier to read 5ebe7b258

Reply:
Indeed would be easier to read 5ebe7b258